Code optimization; new functions

kaushalmodi opened the pull request:
- Use of `funcall` to call the right command based on if the user uses
  `pdf-tools` package or not
- Refine `interleave--doc-view-kill-proc-and-buffer` fn so that the
  `doc-view-kill-proc` is called only if the major mode is
  `doc-view-mode`.
- Added `*interleave--pdf-buffer*`; renamed _interleave--org-buf_
  to _interleave--org-buffer_
- Use of `switch-to-buffer` and `switch-to-buffer-other-window` instead
  of `(other-window 1)` for more accuracy (in the event the user has
  opened a buffer other than the pdf or org notes file in the "other"
  window).
- Added commands to synchronize the pdf page with the page displayed in
  current org notes buffer, to sync the pdf page to the next/previous
  notes page:
  - interleave--sync-pdf-page-current
  - interleave--sync-pdf-page-previous
  - interleave--sync-pdf-page-next
- Creates variable handles `interleave-map` and
  `interleave-pdf-mode-map` so a user can customize the key bindings in
  them without having to modify the source.
- Changed the lighter strings for both pdf and org buffer minor modes to " ≡"

kaushalmodi committed:
Code optimization; new functions

- Use of `funcall` to call the right command based on if the user uses
  `pdf-tools` package or not
- Refine `interleave--doc-view-kill-proc-and-buffer` fn so that the
  `doc-view-kill-proc` is called only if the major mode is
  `doc-view-mode`.
- Added `*interleave--pdf-buffer*`; renamed *interleave--org-buf*
  to *interleave--org-buffer*
- Use of `switch-to-buffer` and `switch-to-buffer-other-window` instead
  of `(other-window 1)` for more accuracy (in the event the user has
  opened a buffer other than the pdf or org notes file in the "other"
  window).
- Added commands to synchronize the pdf page with the page displayed in
  current org notes buffer, to sync the pdf page to the next/previous
  notes page:
  - interleave--sync-pdf-page-current
  - interleave--sync-pdf-page-previous
  - interleave--sync-pdf-page-next
- Creates variable handles `interleave-map` and
  `interleave-pdf-mode-map` so a user can customize the key bindings in
  them without having to modify the source.

kaushalmodi commented on interleave.el:
```diff
@@ -38,28 +39,48 @@
+(if (featurep 'pdf-view) ; if `pdf-tools' is installed
+    (progn
+      ;; Function wrapper for the macro `pdf-view-current-page'
+      (defconst interleave--pdf-current-page-fn                 (lambda () (pdf-view-current-page)))
```
@rudolfochrist  This was the reason why your earlier approach to use the change page hook in pdf-view failed. But I realized that at the very end after I changed everything in this commit. 

If you had wrapped `pdf-view-current-page` in a `lambda` wrapper, your approach would have worked too. But I did not know that then.

kaushalmodi commented:
I need to understand why I see this note on the pull request page: "Can't automatically merge. Don't worry, you can still create the pull request."

kaushalmodi commented on interleave.el:
```diff
@@ -2,7 +2,8 @@
 
 ;; Author: Sebastian Christ <rudolfo.christ@gmail.com>
 ;; URL: https://github.com/rudolfochrist/interleave
-;; Version: 0.2.0
```
The pull request is comparing against a slightly older version. So I am missing some `git` step. Let me know if this is obvious to you. What do I need to do once you have merged the pull request and both of our "master"s are in sync? Do I need to hard reset my fork to your master (How?)?

rudolfochrist replied:
I think rebasing to upstream/master will do the trick.

kaushalmodi committed:
Fix: Wrap `doc-view-current-page' in `lambda' too

- Other minor changes
  - Documentation fill column reduced
  - Slight rewording of Commentary and Usage sections

rudolfochrist commented on interleave.el:
```diff
@@ -118,139 +181,146 @@ to PAGE. It narrows the subtree when found."
+    (save-excursion
+      (org-narrow-to-subtree)
+      (goto-char (point-min))
+      (search-forward-regexp "Notes for page \\(.*\\)")
```
Can you search for the `interleave_page_note` property instead. The headline should be changeable. Thanks.

kaushalmodi committed:
Improve: Search for org heading property

kaushalmodi commented:
Changed the search to org heading properties instead of `"Notes for page"` string in https://github.com/kaushalmodi/interleave/commit/3fff9d9001989a5ae9bf2666bf6c32ceef930ff3

kaushalmodi committed:
Improve: Robustness of `interleave--sync*` fns

- Make the narrowing and widening of org subtrees consistent
- Don't throw error when trying to find the next or previous page notes

rudolfochrist commented:
Works like a charm. Especially the buffer synchronization is nice enhancement. Thank you.
